classes/util/UtilMethods.php

```diff
@@ -1,306 +1,314 @@
+		);
 
-    /**
-     * Returns the shipping method orderline for order
-     * @param $order
-     * @param $products
-     * @param $returnRefundOrderLines
-     * @return array|bool
-     */
-    private function getShippingDetails($order, $products, $returnRefundOrderLines)
-    {
-        //Get the shipping method
-        $orderShippingMethods = $order->get_shipping_methods();
-        $shippingID = 'NaN';
-        $shippingDetails = array();
+		return $compensationAmountOrderline;
+	}
 
-        foreach ($orderShippingMethods as $orderShippingKey => $orderShippingMethods) {
-            $shippingID = $orderShippingMethods['method_id'];
-        }
-        //In a refund it's possible to have order_shipping == 0 and order_shipping_tax != 0 at the same time
-        if ($order->get_shipping_total() <> 0 || $order->get_shipping_tax() <> 0) {
-            if (!empty($products)) {
-                if (!in_array($shippingID, $products['skuList'])) {
-                    return false;
-                }
-            }
-            //getting shipping total and tax applied on it
-            $totalShipping = $order->get_shipping_total();
-            $totalShippingTax = $order->get_shipping_tax();
+	/**
+	 * Returns the shipping method orderline for order
+	 *
+	 * @param WC_Order $order
+	 * @param array    $products
+	 * @param bool     $returnRefundOrderLines
+	 * @return array|bool
+	 */
+	private function getShippingDetails( $order, $products, $returnRefundOrderLines ) {
+		 // Get the shipping method
+		$orderShippingMethods = $order->get_shipping_methods();
+		$shippingID           = 'NaN';
+		$shippingDetails      = array();
 
-            //This will trigger in case a refund action is performed
-            if ($returnRefundOrderLines) {
-                $shippingDetails[$orderShippingKey] = array(
-                    'qty' => 1,
-                    'refund_total' => wc_format_decimal($totalShipping),
-                    'refund_tax' => wc_format_decimal($totalShippingTax)
-                );
-            } else {
-                $shippingDetails[] = array(
-                    'description' => $order->get_shipping_method(),
-                    'itemId' => $shippingID,
-                    'quantity' => 1,
-                    'unitPrice' => number_format($totalShipping, 2, '.', ''),
-                    'taxAmount' => number_format($totalShippingTax, 2, '.', ''),
-                    'taxPercent' => number_format(($totalShippingTax / $totalShipping) * 100, 2, '.', ''),
-                    'goodsType' => 'shipment'
-                );
-            }
-        }
-        return $shippingDetails;
-    }
+		foreach ( $orderShippingMethods as $orderShippingKey => $orderShippingMethods ) {
+			$shippingID = $orderShippingMethods['method_id'];
+		}
+		// In a refund it's possible to have order_shipping == 0 and order_shipping_tax != 0 at the same time
+		if ( $order->get_shipping_total() <> 0 || $order->get_shipping_tax() <> 0 ) {
```

Comment:
Use normal not is, like in the comment...
```suggestion
		if ( $order->get_shipping_total() != 0 || $order->get_shipping_tax() != 0 ) {
```